src/ray/object_manager/object_directory.cc

```diff
@@ -45,26 +48,26 @@ void ObjectDirectory::RegisterBackend() {
       gcs::AsyncGcsClient *client, const ObjectID &object_id,
       const std::vector<ObjectTableDataT> &location_history) {
     // Objects are added to this map in SubscribeObjectLocations.
-    auto object_id_listener_pair = listeners_.find(object_id);
+    auto it = listeners_.find(object_id);
     // Do nothing for objects we are not listening for.
-    if (object_id_listener_pair == listeners_.end()) {
+    if (it == listeners_.end()) {
       return;
     }
     // Update entries for this object.
-    std::vector<ClientID> client_id_vec =
-        UpdateObjectLocations(object_id_listener_pair->second.current_object_locations,
-                              location_history, gcs_client_->client_table());
+    UpdateObjectLocations(location_history, gcs_client_->client_table(),
+                          &it->second.current_object_locations, &it->second.created);
```

Comment:
Same with the other place where we call `UpdateObjectLocations`

Reply:
I think I'm going to move to what Eric and I discussed above about just checking if the object is lost if it's a duplicate, but just to be clear, the past object locations are relevant since `UpdateObjectLocations` processes a subset of the log, not necessarily the whole log.

Reply:
Hm, ok, just for my own clarification, the `location_history` argument in the object notification callback to the object table subscribe function contains the **full history** of all object table updates for that object ID, right?

If that's true, then it looks to me like `UpdateObjectLocations processes the full log, am I missing something?

Reply:
No, it's only the *new* object table updates. So the first notification will contain the full history, but subsequent notifications will only contain a subset.